.github/workflows/publish-containers.yml

```diff
@@ -0,0 +1,113 @@
+name: Publish docker image
+
+on:
+  push:
+    tags:
+      # Matches stable and pre-releases
+      - "[0-9]+.[0-9]+.[0-9]+*"
+    branches:
+      - main
+      - ci/**
+
+jobs:
+  docker:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Outputs the name of the repository (owner/repo)
+      - name: Get image name
+        id: image
+        run: |
+          IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
+          echo ::set-output name=image_name::ghcr.io/${IMAGE_NAME}
+
+      # Tags stable versions as :latest
+      # Pre-releases, alphas, etc. as :snapshot
+      - name: Output image tags from git tag events
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          set -x
+          # Remove everything else than the tagged version
+          VERSION=${GITHUB_REF#refs/tags/}
+          # Tag as stable (:latest) if there is no letters in the version
+          # Otherwise, tag as preview (:snapshot)
+          if [[ $VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            # Matches 1.0.0, 1.0.1, etc.
+            NAMED_VERSION="latest"
+          else
+            # 1.0.0a1, 1.0.0rc1, etc.
+            NAMED_VERSION="snapshot"
+          fi
+          TAGS=$"\
+          ${{ steps.image.outputs.image_name }}:${NAMED_VERSION},\
+          ${{ steps.image.outputs.image_name }}:${VERSION}\
+          "
+          echo "
+          CONTAINER_TAGS=${TAGS}
+          NAMED_VERSION=${NAMED_VERSION}
+          " >> "${GITHUB_ENV}"
+
+      - name: Output image tags from git branch events
+        if: ${{ startsWith(github.ref, 'refs/heads/') }}
+        run: |
+          set -x
+          # Version name is the branch name, slashes are substituted by dashes
+          CLEAN_BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | tr / -)
+          # Target with the branch name as the tag, e.g. unstable-main
+          TAGS=$"\
+          ${{ steps.image.outputs.image_name }}:unstable-${CLEAN_BRANCH_NAME}
+          "
+          # Set version name for open-containers version label as:
+          # <branch>-<drift>-g<6-digit-hash>
+          NAMED_VERSION=$(git describe --all --long | tr -d $'\n')
+          echo "
+          CONTAINER_TAGS=${TAGS}
+          NAMED_VERSION=${NAMED_VERSION}
+          " >> "${GITHUB_ENV}"
```

Comment:
hmm, I think we don't need to build every branch as a docker image unless it's main. Main branch is sort of a dev one for us (trunk based) so the docker image makes sense as a dev preview, however anything else, not relly atm. We do create many branches on a daily basis so it will end up with sort of spam.

Perhaps we can just simply add the condition to gather only the main branch?

Reply:
hmm but you actually restricted that to benches ci/** - ok let's keep that

Reply:
Yes it is only for branches that start with `ci/`. I have followed the practices from the saleor/saleor project.
In my forked repo, you can see how and when it builds: https://github.com/tofran/saleor-dashboard/actions/workflows/publish-containers.yml